src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -164,6 +166,10 @@ public function setContentDisposition($disposition, $filename = '', $filenameFal
         if ('' === $filenameFallback && (!preg_match('/^[\x20-\x7e]*$/', $filename) || false !== strpos($filename, '%'))) {
             $encoding = mb_detect_encoding($filename, null, true);
```

Comment:
This would indeed help with generating the filenameFallback no matter how weird the $filename is. But considering `ResponseHeaderbag::makeDisposition` the `$filename` parameter is described and expected to be `A unicode string` and placed in the header field parameter `filename*` defined as utf-8 encoded without further processing. If passing e.g. an ISO-8859-1 encoded $filename with german umlauts and not throwing this exception would lead to a bad or invalid Content-Disposition response header. Maybe `setContentDisposition` should actually ensure the $filename to be utf-8 encoded in any case. But this could be a BC break, although I think everything but utf-8 seems to be a misusage. What do you think?